app/src/main/java/io/github/sheasmith/discountdetective/viewmodel/ShoppingListViewModel.kt

```diff
@@ -43,9 +43,20 @@ class ShoppingListViewModel @Inject constructor(
     /**
      * The current shopping list items
      */
-    val shoppingList: MutableStateFlow<List<Triple<RetailerProductInformation, StorePricingInformation, Int>>?> =
+    val shoppingList: MutableStateFlow<List<Triple<RetailerProductInformation, StorePricingInformation, ShoppingListItem>>?> =
         MutableStateFlow(null)
 
+    /**
+     * Update the state of the checkbox in the shoppingList
+     * by copy then delete and insert so compose tracks changes
+     */
+    fun changeShoppingListChecked(item: ShoppingListItem, checked: Boolean) {
```

Comment:
If you set `.checked` inside of the view (e.g. in the onCheckedChanged callback), then you will only need one parameter here.

Reply:
fair, but my current understanding of callbacks is quite limited